homeassistant/components/rainmachine/config_flow.py

```diff
@@ -56,9 +57,15 @@ def async_get_options_flow(
 
     async def async_step_homekit(self, discovery_info: DiscoveryInfoType) -> FlowResult:
         """Handle a flow initialized by homekit discovery."""
-        return await self.async_step_zeroconf(discovery_info)
+        return await self.async_step_homekit_zeroconf(discovery_info)
 
     async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle discovery via zeroconf."""
+        return await self.async_step_homekit_zeroconf(cast(dict, discovery_info))
```

Comment:
Once homekit is also migrated to use ZeroconfServiceInfo (or sub class) we can remove the cast.